src/main/java/edu/umich/its/cpm/MigrationInstanceService.java

```diff
@@ -8,27 +8,104 @@
 
 import org.springframework.scheduling.annotation.Async;
-import org.springframework.scheduling.annotation.AsyncResult;
 import org.springframework.stereotype.Service;
 import org.springframework.util.StopWatch;
 
 @Service
 @Component
-public class MigrationInstanceService{
-
+public class MigrationInstanceService {
+	
 	private static final Logger log = LoggerFactory
 			.getLogger(MigrationInstanceService.class);
 	
 	@Autowired
-	MigrationTaskService migrationTaskService;
+	private MigrationTaskService migrationTaskService;
+	
+	@Async
+	public void runProcessingThread() throws InterruptedException {
+		
+		log.info("Box Migration Processing thread is running");
+		
+		List<Future<HashMap<String, String>>> futureList = new ArrayList<Future<HashMap<String, String>>>();
+
+		log.info("Box Migration Processing thread is running01");
+		while (true)
+		{
+
+			log.info("Box Migration Processing thread is running02");
+			 // Artificial delay of 5s for demonstration purposes
+			Thread.sleep(5000L);
+	        
+			// looping through
+			HashMap<String, LinkedList<MigrationFields>> boxMigrationRequests = BoxUtils.getBoxMigrationRequests();
+			Iterator<String> iUserIds = boxMigrationRequests.keySet().iterator();
+			while (iUserIds.hasNext())
+			{
+				String userId = iUserIds.next();
+				// get the migration request queue belongs to the user
+				LinkedList<MigrationFields> lList = boxMigrationRequests.get(userId);
+				
+				MigrationFields mFields = lList.get(0);
+				if (!mFields.getProcessed())
+				{
+					log.info("Box migration request processing user " + userId + " request queue length=" + lList.size());
+					
+					try
+					{
+						// if the first request for the user is not processed yet
+						// process with the Box upload request
+						futureList.add( migrationTaskService.uploadToBox(mFields.getEnv(), mFields.getRequest(), mFields.getResponse(), mFields.getUserId(), mFields.getSessionAttributes(), mFields.getSiteId(), mFields.getBoxFolderId(), mFields.getMigrationId(), mFields.getRepository()));
+						
+						// mark this request as being processed
+						mFields.setProcessed(true);
+						lList.set(0, mFields);
+						BoxUtils.setBoxMigrationRequestForUser(userId, lList);
+					}
+					catch (java.lang.InterruptedException e)
+					{
+						log.error("Error processing Box upload request from user " + userId + " for site " + mFields.getSiteId() + e.getMessage());
+					}
+				}	// if
+			}	// while
+		    
+			boxMigrationRequests = BoxUtils.getBoxMigrationRequests();
+			// get a cloned list, in case we need to remove the finished async tasks from the original list
+			List<Future<HashMap<String, String>>> futureListClone = new ArrayList<Future<HashMap<String, String>>>();
+			futureListClone.addAll(futureList);
+			
+			for (Future<HashMap<String, String>> future : futureListClone) {
+	            try {
+					// get the status of asynchronize processed Box upload request
+					HashMap<String, String> rv = future.get();
```

Comment:
I've not used Future in Java so I'm not clear on all the details.  A get call on a future will block and wait for the computation to finish. A status check can be done with isDone. Should the code be waiting or checking?  I.e. is the comment or the code wrong?